docs/user-documentation.rst

```diff
@@ -0,0 +1,425 @@
+    ...
+    if operator.is_active('foo'):
+        ... do something ...
+    else:
+        ... do something else ...
+
+If autocreate is on (and it is by default), the 'foo' switch will be
+automatically created and set to disabled the first time it is referenced.
+Activating the switch and controlling exactly when the switch is active,
+are covered in `Managing switches`_.
+
+In templates
+------------
+
+Every templating framework has its own take on how (or even if) logic may be
+used. That said, Switchboard provides some helpers to make things easier. If
+
+In javascript
+-------------
+
+The easiest way to use Switchboard in conjunction with Javascript is to set a
+Javascript flag within your template code and then use that flag within your
+Javascript logic. Using Mako's syntax in the template::
+
+    <%!
+        from switchboard import operator
+    %>
+    <script>
+        window.switches = window.switches || {};
+        % if operator.is_active('foo'):
+        switches.foo = true;
+        % else:
+        switches.foo = false;
+        % endif
+    </script>
+
+In the Javascript::
+
+    if (switches.foo) {
+        ... do something ...
+    } else {
+        ... do something else ...
+    }
+
+Again, this time using Jinja syntax and the Switchboard-provided "active"
+test_::
+
+    <script>
+        window.switches = {};
+        switches.foo = {{ 'true' if 'foo' is active else 'false' }};
+    </script>
+
+Context objects
+---------------
+
+Every switch is evaluated (to see if it is active or not) within a particular
+context. By default, that context includes the request object, which allows
+Switchboard to specify conditions such as: "make this switch active only for
+requests with 'foo' in the query string." That said, there may be other
+objects that would be handy to have available in the context. For example, in
+an ecommerce setting, the Product model may have a ``new`` flag. By passing
+the model into the ``is_active`` method, Switchboard can now activate
+switches based on that flag::
+
+    if operator.is_active('foo', my_product):
+
+Any objects passed into the ``is_active`` method after the switch's key will be
+added to the context. Normally when dealing with context objects, a custom
+condition will be required to actually evaluate the switch against that object.
+
+Custom conditions
+-----------------
+
+Switchboard supports custom conditions, allowing application developers to
+adapt switches to their particular needs. Creating a condition typically
+consists of extending `switchboard.conditions.ConditionSet`.
+
+An example: if the application needs to activate switches for visitors from a
+particular country, a custom condition can do the geo lookup on the IP from
+the request and return the country value::
+
+    from switchboard.conditions import ConditionSet, Regex
+    from my_app.geo import country_code_by_addr
+
+    class GeoConditionSet(ConditionSet):
+        countries = Regex()
```

Comment:
It may be nicer to pare the example of non-contributing parts. This countries value and the Regex are just distracting since they are not used further in the example.